freqtrade/freqtradebot.py

```diff
@@ -102,6 +106,22 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self._sell_lock = Lock()
         LoggingMixin.__init__(self, logger, timeframe_to_seconds(self.strategy.timeframe))
 
+        if self.config.get("trading_mode"):
+            self.trading_mode = TradingMode(self.config.get("trading_mode"))
```

Comment:
One important thing here:
if we use new setting keys from the config - these should be added to `configuration.md` - as well as to `constants.py` so we can have some reliance on the type we expect (based on this - i have no idea what `trading_mode` should/could be set to without following the code flow.
I'd also add their default to the full config sample - so it's easy to copy/paste.

---

Just to throw this into the room - we seem to be having "quite some" settings in this regard ... so i think we should try to group them (all leverage/short settings into one subsetting) so they are logically grouped.

It might even make sense to move them into the exchange substructure (but this might need further consideration) - as in my understanding, the available settings will heavily depend on the selected exchange (one mode might work on binance, but not on kraken and viceversa).